examples/oneapi/cpp/source/misc/read_batch.cpp

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/io/csv.hpp"
+
+#include "example_util/utils.hpp"
+
+namespace dal = oneapi::dal;
+template <typename TableReadType>
+void run() {
+    const auto data_file_name = get_data_path("covcormoments_dense.csv");
+
+    const auto table =
+        dal::read<dal::table>(dal::csv::data_source<TableReadType>{ data_file_name });
+
+    const auto type = table.get_metadata().get_data_type(0);
+    switch (type) {
+        case dal::data_type::float64: std::cout << "table data type: double " << std::endl; break;
+        case dal::data_type::float32: std::cout << "table data type: float " << std::endl; break;
+        default: std::cout << "table data type: null " << std::endl; break;
```

Comment:
I think there should be an exception that might show that something goes wrong with the example. The same applies to GPU example.

Reply:
Not sure what did you mean here. If you are referring to invalid data type usage, then it is gonna fail during compile due to the way static_asserts work. If it is anything else, could you please provide more details?

Reply:
I think it would be better just to remove printing in the `default` case then.